pkg/common/spi/provider.go

```diff
@@ -114,4 +114,10 @@ type Provider interface {
 
 	// Upgrade requests the provider initiate a cluster upgrade to the given version
 	Upgrade(clusterID string, version string, t time.Time) error
+
+	// GetUpgradePolicy gets the first upgrade policy from the top
+	GetUpgradePolicy(clusterID string) (string, error)
```

Comment:
Very minor nit, but maybe this (and its implementations) might be better named `GetUpgradePolicyID` since it's returning that specific element.. at least for the OCM provider.